### PR TITLE
TST: Test dimensions/indices found from parsed gufunc signatures.

### PR DESCRIPTION
**Pull request by mhvk**
Another split-off from #11132 - expanded signature testing.

### CODE REVIEW

File: numpy/core/tests/test_ufunc.py [hunk 1]
```diff
@@ -327,7 +333,10 @@ def test_signature(self):
             pass
 
         # more complicated names for variables
-        assert_equal(umt.test_signature(2, 1, "(i1,i2),(J_1)->(_kAB)"), 1)
+        enabled, num_dims, ixs = umt.test_signature(2, 1, "(i1,i2),(J_1)->(_kAB)")
+        assert_equal(enabled, 1)
+        assert_equal(num_dims, (2, 1, 1))
+        assert_equal(ixs, (0, 1, 2, 3))
```
Comment (eric-wieser) [inline numpy/core/tests/test_ufunc.py]: Might be nicer to build a nested tuple here, and I don't envisage it being any harder

**Review by eric-wieser — APPROVED** [review]
One suggestion, but fine to merge as is

**Comment by mhvk** [comment]
I think for the testing purposes here it is perhaps better to just give back what the ufunc contains. I have been wondering more generally, though, about some way to expose this information -- maybe as simple as a nicer `__repr__` -- and then I think it would be good to distinguish more clearly.

**Comment by mhvk** [comment]
@mattip - are you happy with this going in separately? (I do apologize for hijacking your PR...)

**Comment by eric-wieser** [comment]
> for the testing purposes here it is perhaps better to just give back what the ufunc contains

Arguably the ufunc just contains a representation of nested tuple, split between `num_dims` and `idx`. So I'd maybe just return `(core_enabled, tuple_of_idx)`.

I don't feel strongly about this - looks like a good thing to test however you spell it

**Comment by mattip** [comment]
LGTM, breaking the larger work into a number of smaller PRs is the way to go

**Pull request merged**